src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -104,11 +104,14 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         return this._identity;
     }
     public get notebookUri(): Uri | undefined {
-        return this.notebookDocument?.uri;
+        return this._notebookDocument?.uri;
     }
     public get notebookEditor(): NotebookEditor | undefined {
         return this._notebookEditor;
     }
+    public get notebookDocument(): NotebookDocument | undefined {
```

Comment:
Exposed a property `notebookDocument`, we have a lot of `notebookUri.toString()` comparisons of strings, instead of just comparing the object refs.